### PR TITLE
Added a search on the server side.

### PR DESCRIPTION
**Pull request by ostcar**
Please review and test this PR. There could be erros in it but also some open design decisions.

Use localhost:8000/search/?q=foo to search for foo 

Stil open:
- [ ] manage command to recreate the index with existing instances
- [x] support umlauts in the search query string
- [x] isort and flake8

### CODE REVIEW

**Comment by ostcar** [comment]
Hit to myseld. The speakers have to be indexed to each related item

**Comment by emanuelschuetze** [comment]
Tested:
I created a motion and added a submitter "Max Mustermann". The search finds the motion if I search for "max" or "mustermann" but not for "muster". 
=> Search should also work for substrings.

File: openslides/core/views.py [hunk 1]
```diff
@@ -561,3 +563,17 @@ def get_context_data(self, **context):
                 'description': get_plugin_description(plugin),
                 'version': get_plugin_version(plugin)})
         return result
+
+
+class SearchView(utils_views.APIView):
+    """
+    Accepts a search string and returns a list of objects where each object
+    is a dictonary with the keywords collection and id.
+    """
```
Comment (normanjaeckel) [inline openslides/core/views.py]: Proposal: Add a link to the search string syntax.

File: openslides/utils/search.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+    def create_index(self):
+        """
+        Creats the whoosh index. Delets an existing index if exists.
+
+        Returns the index.
+        """
+        path = self.get_index_path()
+        if path == 'ram':
+            self.storage = RamStorage().create_index(get_schema())
```
Comment (normanjaeckel) [inline openslides/utils/search.py]: Proposal: Add a comment that this case is only for testing.

File: openslides/utils/models.py [hunk 3]
```diff
@@ -39,3 +39,11 @@ def get_root_rest_url(self):
         root_instance = self.get_root_rest_element()
         rest_url = '%s-detail' % type(root_instance)._meta.object_name.lower()
         return reverse(rest_url, args=[str(root_instance.pk)])
+
+    def get_collection_string(self):
+        """
+        Returns the string representing the name of the collection.
+        """
+        # TODO: find a way not to use the url
```
Comment (normanjaeckel) [inline openslides/utils/models.py]: Please add an issue and remove this TODO.

File: openslides/core/views.py (hunk 1 above)
Comment (ostcar) [inline openslides/core/views.py]: done

File: openslides/utils/models.py (hunk 3 above)
Comment (ostcar) [inline openslides/utils/models.py]: #1791

File: openslides/utils/search.py (hunk 2 above)
Comment (ostcar) [inline openslides/utils/search.py]: See #1792

**Pull request merged**